src/braket/default_simulator/state_vector_simulation.py

```diff
@@ -155,6 +149,13 @@ def state_with_observables(self) -> np.ndarray:
             raise RuntimeError("No observables applied")
         return self._post_observables
 
+    def expectation(self, observable: Observable) -> float:
+        qubit_count = self._qubit_count
+        with_observables = observable.apply(np.reshape(self._state_vector, [2] * qubit_count))
```

Comment:
Great! Much more readable, and opens the door to future optimization if we want to special case `apply` for specific observables.